nixpkgs-docs: when to prefer passthru.tests over installCheckPhase (#119731)
NixOS/nixpkgs: doc/stdenv/stdenv.chapter.md

The installCheck phase checks whether the package was installed correctly by running its test suite against the installed directories. The default `installCheck` calls `make installcheck`.

> If you want to add additional tests that are not part of the source distribution, it is better to add them to `passthru.tests` (see <xref linkend="var-meta-tests"/>) instead. That avoids adding overhead to every build, and allows the tests to be executed independently of the build.

Reviewer: ```suggestion
It is better to add tests that are not part of the source distribution to `passthru.tests` (see <xref linkend="var-meta-tests"/>) instead. This avoids adding overhead to every build and enables us to run the tests independently.
```

The installCheck phase checks whether the package was installed correctly by running its test suite against the installed directories. The default `installCheck` calls `make installcheck`.

Because this adds overhead to every build, it is usually recommended to add the test to

> <xref linkend="var-meta-tests"/> instead.

Reviewer 2: My 2c is that I think installCheck is primarily useful for applications that have test-suites in tree (that maybe rely on build system configuration artifacts), that aren't then shipped as part of the package build.

If you as a package maintainer are defining a test, then it probably does make sense to add it as a separate passthru.tests test.

Author: that's a great distinction, updated to make that more clear
